client/my-sites/plugins/plugin-details.jsx

```diff
@@ -0,0 +1,440 @@
+import { isBusiness, isEcommerce, isEnterprise } from '@automattic/calypso-products';
+import { Button } from '@automattic/components';
+import classNames from 'classnames';
+import { useTranslate } from 'i18n-calypso';
+import page from 'page';
+import { useEffect, useMemo } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import DocumentHead from 'calypso/components/data/document-head';
+import QueryJetpackPlugins from 'calypso/components/data/query-jetpack-plugins';
+import EmptyContent from 'calypso/components/empty-content';
+import FixedNavigationHeader from 'calypso/components/fixed-navigation-header';
+import { useLocalizedMoment } from 'calypso/components/localized-moment';
+import MainComponent from 'calypso/components/main';
+import PageViewTracker from 'calypso/lib/analytics/page-view-tracker';
+import formatNumberCompact from 'calypso/lib/format-number-compact';
+import { userCan } from 'calypso/lib/site/utils';
+import PluginNotices from 'calypso/my-sites/plugins/notices';
+import { isCompatiblePlugin } from 'calypso/my-sites/plugins/plugin-compatibility';
+import PluginSections from 'calypso/my-sites/plugins/plugin-sections';
+import PluginSectionsCustom from 'calypso/my-sites/plugins/plugin-sections/custom';
+import PluginSiteList from 'calypso/my-sites/plugins/plugin-site-list';
+import { siteObjectsToSiteIds } from 'calypso/my-sites/plugins/utils';
+import SidebarNavigation from 'calypso/my-sites/sidebar-navigation';
+import { recordGoogleEvent } from 'calypso/state/analytics/actions';
+import {
+	getPluginOnSite,
+	getPluginOnSites,
+	getSiteObjectsWithPlugin,
+	getSitesWithoutPlugin,
+	isRequestingForSites,
+} from 'calypso/state/plugins/installed/selectors';
+import { removePluginStatuses } from 'calypso/state/plugins/installed/status/actions';
+import { fetchPluginData as wporgFetchPluginData } from 'calypso/state/plugins/wporg/actions';
+import {
+	isFetching as isWporgPluginFetching,
+	isFetched as isWporgPluginFetched,
+	getPlugin as getWporgPlugin,
+} from 'calypso/state/plugins/wporg/selectors';
+import { canCurrentUser } from 'calypso/state/selectors/can-current-user';
+import canCurrentUserManagePlugins from 'calypso/state/selectors/can-current-user-manage-plugins';
+import getSelectedOrAllSitesWithPlugins from 'calypso/state/selectors/get-selected-or-all-sites-with-plugins';
+import {
+	isJetpackSite as checkJetpackSite,
+	isRequestingSites as checkRequestingSites,
+} from 'calypso/state/sites/selectors';
+import { getSelectedSite, getSelectedSiteId } from 'calypso/state/ui/selectors';
+import NoPermissionsError from './no-permissions-error';
+
+function PluginDetails( props ) {
+	const dispatch = useDispatch();
+	const moment = useLocalizedMoment();
+	const translate = useTranslate();
+
+	const selectedSiteId = useSelector( getSelectedSiteId );
```

Comment:
we can use `selectedSite.ID` directly since we 've got `const selectedSite = useSelector( getSelectedSite );`